2015-Fall/mhartvig/Calculator/Calculator/ViewController.swift

```diff
@@ -79,12 +89,16 @@ class ViewController: UIViewController {
     
     @IBAction func Clear() {
         brain.clear()
+        clearDisplay()
+    }
+    
+    func clearDisplay() {
         displayValue = 0
         isUserTyping = false
         isNumberADecimal = false;
     }
     
-    @IBAction func dotPress() {
+    @IBAction func dotPress() {        
         if !isNumberADecimal {
         if (isUserTyping) {
             display.text = display.text! + ",";
```

Comment:
Bonus info. To localize the dot you can do `NSNumberFormatter().decimalSeparator` it will return the decimal seperator w.r.t. the locale used on the device.